pa11ycrawler/spiders/edx.py

```diff
@@ -0,0 +1,94 @@
+"""
+A spider that can crawl an Open edX instance.
+"""
+import os
+from datetime import datetime
+from urlobject import URLObject
+from scrapy.spiders import CrawlSpider, Rule
+from scrapy.linkextractors import LinkExtractor
+from pa11ycrawler.items import A11yItem
+
+AUTO_AUTH_PATH = "/auto_auth"
+COURSE_BLOCKS_API_PATH = "/api/courses/v1/blocks"
+
+
+class EdxSpider(CrawlSpider):
+    "A Scrapy spider that can crawl an Open edX instance."
+    name = 'edx'
+
+    rules = (
+        Rule(
+            LinkExtractor(
+                deny=[
+                    # don't crawl logout links
+                    r"/logout/",
+                    # don't crawl xblock links
+                    r"://[^/]+/xblock/",
+                    # don't crawl anything that returns an archive
+                    r"\?_accept=application/x-tgz",
+                ],
+                unique=True,
+            ),
+            callback='parse_item',
+            follow=True,
+        ),
+    )
+
+    def __init__(
+            self,
+            domain="localhost",
+            port="8000",
+            course_key="course-v1:edX+Test101+course",
+            data_dir="data",
+        ):  # noqa
+        self.allowed_domains = [domain]
+        port = int(port)
+
+        # set start URL based on course_key, which is the test course by default
+        api_url = (
+            URLObject("http://")
```

Comment:
This is just the starting URL. On production, `http://` URLs are redirected to `https://`, and the crawler will follow those redirects appropriately. When running on localhost, setting up `https://` is a pain, so it's nicer to start with `http://` instead.